osp/core/session/sparql_backend.py

```diff
@@ -72,8 +71,7 @@ def __enter__(self):
 
     def __call__(self, **kwargs):
         """Add kwargs to datatypes when class is called."""
-        self.datatypes = kwargs
-        return self.__iter__()
+        return self.__iter__(**kwargs)
```

Comment:
@kysrpex: the simplest idea was just simply passing the `kwargs` to the `__iter__` function of the `SparqlResult`. This leads to the expected behaviour [you proposed](https://github.com/simphony/osp-core/pull/663#issuecomment-873980500).

Reply:
This has the drawback of breaking existing implementations (e.g. the Allegrograph) and requiring more attention from the wrapper developer. I finally managed to implement things without requiring changes on the wrappers. See  [81a2005](https://github.com/simphony/osp-core/pull/663/commits/81a20056676caf3d537e4bf26b81b06e1d511b6e).